pkspace/utils/loaders.py

```diff
@@ -0,0 +1,105 @@
+import glob
+import json
+import os
+import xml.etree.ElementTree
+import cv2
+import numpy as np
+
+
+class Loader:
+    def load(self, path, split):
+        pass
+
+
+class PKspaceLoader(Loader):
+    def __get_spaces__(self, src, desc):
+        finsize = (80, 80)
+        spaces = []
+        answers = []
+        for space in desc.get('spots'):
+            points = space.get('points')
+            angle = space.get('rotation')
+            nove_pole = np.hstack(
+                (np.asarray(points), np.ones((len(points), 1))))
+            center = np.mean(nove_pole, 0)[:2]
+            center = (center[0], center[1])
+            rows, cols, tmp = src.shape
+            m = cv2.getRotationMatrix2D(center, -angle, 1)
+            nove_pole = m.dot(nove_pole.T).T
+            nove_pole[:, 0] = np.clip(nove_pole[:, 0], 0, cols)
+            nove_pole[:, 1] = np.clip(nove_pole[:, 1], 0, rows)
+            minima = nove_pole.min(axis=0)
+            maxima = nove_pole.max(axis=0)
+            min_x, min_y = int(minima[0]), int(minima[1])
+            max_x, max_y = int(maxima[0]), int(maxima[1])
+            rotated = cv2.warpAffine(src, m, (cols, rows))
+            roi = rotated[min_y:max_y, min_x:max_x]
+            fin = cv2.resize(roi, finsize).flatten()
+            spaces.append(fin)
+            answers.append(space.get('occupied', -1))
+        return spaces, answers
+
+    def load(self, path, split, extension='.png'):
+        all_parking_spaces = []
```

Comment:
@rstevanak what was previously called `spaces` is now called `parking_spaces` -- let's use just one of them for consistency.